source/simulator/assembly.cc

```diff
@@ -779,14 +784,29 @@ namespace aspect
                                               (parameters.stokes_velocity_degree+1)/2),
                                   parameters.n_compositional_fields);
 
+    int count_smoothing = 0;
     typename DoFHandler<dim>::active_cell_iterator cell = dof_handler.begin_active();
     for (unsigned int cellidx=0; cellidx<triangulation.n_active_cells(); ++cellidx, ++cell)
       {
-        if (!cell->is_locally_owned())
-          {
-            viscosity_per_cell[cellidx]=-1;
-            continue;
-          }
+    	cell->set_user_index(count_smoothing);
+    	count_smoothing++;
+    	if (use_smoothing == false)
```

Comment:
try to simplify this logic by starting the for loop with something like
```
if (cell->is_artificial())
{
  viscosity_per_cell[cellidx]=-1;
  continue;
}
cell->set_user_index(count_smoothing);
```
and remove the ```use_smoothing==true``` case here